src/Juvix/Compiler/Core/Evaluator.hs

```diff
@@ -563,3 +573,17 @@ toCoreError loc (EvalError {..}) =
       _coreErrorNode = _evalErrorNode,
       _coreErrorLoc = fromMaybe loc (lookupLocation =<< _evalErrorNode)
     }
+
+-- | A class that provides a function `checkApply` that applies a function to a
+-- list of arguments and fails if the length of the arguments list is not equal
+-- to the arity of the function.
+class CheckApplyType a t where
```

Comment:
I did this because at some point I needed to apply the type of the arguments. But I don't need to do that now and I agree that it's more intuitive the other way.

Reply:
done